extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSFilter.java

```diff
@@ -97,7 +97,10 @@ public void handle(RoutingContext event) {
                 response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, origin);
             }
 
-            response.headers().set(HttpHeaders.ACCESS_CONTROL_ALLOW_CREDENTIALS, "true");
+            boolean allowCredentials = corsConfig.accessControlAllowCredentials
+                    .orElseGet(() -> (corsConfig.origins.isPresent() && !corsConfig.origins.get().contains("*")));
```

Comment:
Can you also have a boolean var recording the result of the origin comparison checked here ? It is probably not very critical but it is better to return `true` only if the exact origin match happened...

Reply:
Done. Thanks :)